File: backend/src/phasor_workbench/api/routes/health.py
"""Liveness endpoint.

The only endpoint implemented during Milestone 1. It exists to satisfy the
milestone's exit criterion that frontend and backend communicate.
"""

from typing import Literal

from fastapi import APIRouter
from pydantic import BaseModel

from phasor_workbench import __version__

router = APIRouter(tags=["system"])


class HealthResponse(BaseModel):
    status: Literal["ok"] = "ok"
    version: str


@router.get("/health", response_model=HealthResponse, summary="Liveness check")
def get_health() -> HealthResponse:
    return HealthResponse(version=__version__)

File: frontend/src/api/health.ts
import { request } from './client';
import type { HealthResponse } from './types';

/**
 * Liveness check. Used to prove frontend and backend can communicate,
 * which is Milestone 1's exit criterion.
 */
export function getHealth(signal?: AbortSignal): Promise<HealthResponse> {
  return request<HealthResponse>('/health', signal ? { signal } : {});
}
